parquet-tools/src/main/java/parquet/tools/command/CatCommand.java

```diff
@@ -52,7 +69,11 @@ public void execute(CommandLine options) throws Exception {
       PrintWriter writer = new PrintWriter(Main.out, true);
       reader = new ParquetReader<SimpleRecord>(new Path(input), new SimpleReadSupport());
       for (SimpleRecord value = reader.read(); value != null; value = reader.read()) {
-        value.prettyPrint(writer);
+        if (options.hasOption('j')) {
+          value.prettyPrintJson(writer);
```

Comment:
Another approach is to use different subtypes of SimpleRecord (ListSimpleRecord, MapSimpleRecord, ...) for this.